rotkehlchen/chain/evm/utils.py

```diff
@@ -0,0 +1,142 @@
+    VELODROME_POOL_PROTOCOL: 'VELO_V2_LP',
+}
+
+
+def find_lp_price(
+        evm_inquirer: 'EvmNodeInquirer',
+        token: EvmToken,
+        token_price_func: Callable,
+        token_price_func_args: list[Any],
+        block_identifier: BlockIdentifier,
+        abi_name: Optional['LP_TOKEN_CONTRACT_ABI_NAMES'] = None,  # the lp token's contract abi that will be used to query the info needed for lp token price calculation  # noqa: E501
+) -> Optional[Price]:
+    """
+    Calculate the price for an LP token. That is:
+    value = (Total value of liquidity pool) / (Current supply of LP tokens)
+    We need:
+    - Price of token 0
+    - Price of token 1
+    - Pooled amount of token 0
+    - Pooled amount of token 1
+    - Total supply of pool token
+    """
+    address = token.evm_address
+    if abi_name is None:
+        abi_name = token_protocol_to_abi_name[token.protocol]
+    contract = EvmContract(
+        address=address,
+        abi=evm_inquirer.contracts.abi(abi_name),
+        deployed_block=0,
+    )
+    methods = ['token0', 'token1', 'totalSupply', 'getReserves', 'decimals']
+    multicall_method = evm_inquirer.multicall_2  # choose which multicall to use
+    if isinstance(block_identifier, int):
+        if block_identifier <= 7929876:
+            log.error(
+                f'No multicall contract at the block {block_identifier}. {abi_name} '
+                f'query failed. Should implement direct queries',
+            )
+            return None
+
+        if block_identifier <= 12336033:
+            multicall_method = evm_inquirer.multicall
+
+    try:
+        output = multicall_method(
+            require_success=True,
+            calls=[(address, contract.encode(method_name=method)) for method in methods],
+            block_identifier=block_identifier,
+        )
+    except RemoteError as e:
+        log.error(
+            f'Remote error calling multicall contract for {abi_name} '
+            f'token {token.evm_address} properties: {e!s}',
+        )
+        return None
+
+    # decode output
+    decoded = []
+    for (method_output, method_name) in zip(output, methods):
+        call_success = True
+        if multicall_method == evm_inquirer.multicall_2:
+            call_success = method_output[0]
+            call_result = method_output[1]
+        else:
+            call_result = method_output  # type: ignore
+        if call_success and len(call_result) != 0:
+            decoded_method = contract.decode(call_result, method_name)
+            if len(decoded_method) == 1:
+                # https://github.com/PyCQA/pylint/issues/4739
+                decoded.append(decoded_method[0])
+            else:
+                decoded.append(decoded_method)
+        else:
+            log.debug(
+                f'Multicall to {abi_name} failed to fetch field {method_name} '
+                f'for token {token.evm_address}',
+            )
+            return None
+
+    chain_id = token.chain_id
```

Comment:
truth is that there is no reason to assign this to a variable. You can access directly `token.chain_id` in the two places where it is used since there is no extra cost on it